easybuild/tools/toolchain/toolchain.py

```diff
@@ -620,6 +649,20 @@ def prepare(self, onlymod=None, silent=False, loadmod=True):
                 self.generate_vars()
                 self._setenv_variables(onlymod, verbose=not silent)
 
+        if build_option('use_compiler_cache'):
+            paths = {}
+            for cache in [CCACHE, F90CACHE]:
+                cache_path = which(cache)
+                if cache_path is None:
+                    raise EasyBuildError("%s binary not found in $PATH", cache)
+                paths[cache] = cache_path
+
+            for cachename in ["CCACHE", "F90CACHE"]:
+                os.environ["%s_DIR" % cachename] = build_option('use_compiler_cache')
```

Comment:
only call out to `build_option` once, so do `compiler_cache = build_option('use_compiler_cache')` above and then use `compiler_cache` instead